openhands/server/listen.py

```diff
@@ -407,6 +414,23 @@ async def list_files(request: Request, path: str | None = None):
         )
     runtime: Runtime = request.state.session.agent_session.runtime
     file_list = runtime.list_files(path)
+    file_list = [f for f in file_list if f not in FILES_TO_IGNORE]
+
+    def filter_for_gitignore(file_list, base_path):
+        gitignore_path = os.path.join(base_path, '.gitignore')
+        try:
+            read_action = FileReadAction(gitignore_path)
+            observation = runtime.run_action(read_action)
+            spec = PathSpec.from_lines(
+                GitWildMatchPattern, observation.content.splitlines()
+            )
+        except Exception as e:
+            print(e)
+            return file_list
+        file_list = [entry for entry in file_list if not spec.match_file(entry)]
+        return file_list
+
+    file_list = filter_for_gitignore(file_list, '')
```

Comment:
The check for the .gitignore actually existing is missing completely. Also, the fallback is gone. Why?

Reply:
The check is now in `try/catch`--if it doesn't exist, we just bail out. I could probably be more explicit there though

The fallback is an antipattern IMO. Hiding a random list of folders that the user might expect to be available is potentially very confusing, and there was a lot in there

The only folder we should hide by default is `.git`

Reply:
Ah interesting. Do you have any record here? Would love to take a look

Hard for me to imagine folks not just adding those to .gitignore...

Reply:
The only PR I could find on short notice is #2621

Reply:
Adding things like `.DS_Store` does make sense to me. But `logs`, `dist`, etc seem too aggressive--I have projects where I check `dist` into the repo for example, and I'd be very confused why it's not there. So I'm curious what folders folks specifically asked to be hidden

Reply:
I didn't mean that each entry in the fallback list was specifically requested to be ignored, just that the "clutter" in the UI should be less. I think back then I first added that hard-coded list and in a follow up PR the .gitignore support, but kept that list as the fallback. Most of those entries were based on _my_ user perspective, so I understand if this list may not be representative for all scenarios, they should be removed.

Reply:
Ahh got it! Thanks for the context. I'll add back in a few of the most obvious defaults. But I think we should keep it very minimal--much better to have clutter than confusion :)